### PR TITLE
setting to disable notifications for new contacts

### LINKED ISSUE #4572: Notification for new contacts
Signal gives you a notification, when it finds someone from your contacts on the service. It is basically a useful feature. But it needs some user control, I think.

For example I prefer that those notifications are silent and without vibration.

### PR DESCRIPTION
**Pull request by agrajaghh**
also mute these notifications between 23:00 and 09:00

fixes #4572

### CODE REVIEW

**Comment by heyimgay** [comment]
Can you add an option for a silent notification? So On/Silence/Off

**Comment by agrajaghh** [comment]
I would be okay with it, its up to @moxie0

File: src/org/thoughtcrime/securesms/util/DirectoryHelper.java [hunk 1]
```diff
@@ -217,7 +217,14 @@ private static void notifyNewUsers(@NonNull  Context context,
       if (!SessionUtil.hasSession(context, masterSecret, newUser)) {
         IncomingJoinedMessage message        = new IncomingJoinedMessage(newUser);
         Pair<Long, Long>      smsAndThreadId = DatabaseFactory.getSmsDatabase(context).insertMessageInbox(message);
-        MessageNotifier.updateNotification(context, masterSecret, smsAndThreadId.second);
+        if (TextSecurePreferences.isNewContactsNotificationEnabled(context)) {
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/util/DirectoryHelper.java]: I was thinking we wouldn't do anything at all if this is disabled.

Comment (agrajaghh) [inline src/org/thoughtcrime/securesms/util/DirectoryHelper.java]: ah okay, I thought its just about the notification. I'll remove the messages as well..

**Comment by agrajaghh** [comment]
@moxie0 adressed your comment

**Comment by moxie0** [comment]
in 3.9.0

**Pull request closed without merging**